spec/API_specification/linear_algebra_functions.md

```diff
@@ -11,6 +11,46 @@ A conforming implementation of the array API standard must provide and support t
 -   Unless stated otherwise, functions must adhere to the type promotion rules defined in {ref}`type-promotion`.
 -   Unless stated otherwise, floating-point operations must adhere to IEEE 754-2019.
 
+## Design Principles
+
+A principal goal of this specification is to standardize commonly implemented interfaces among array libraries. While this specification endeavors to avoid straying too far from common practice, this specification does, with due restraint, seek to address design decisions arising more from historical accident than first principles. This is especially true for linear algebra APIs, which have arisen and evolved organically over time and have often been tied to particular underlying implementations (e.g., to BLAS and LAPACK).
+
+Accordingly, the standardization process affords the opportunity to reduce interface complexity among linear algebra APIs by inferring and subsequently codifying common design themes, thus allowing more consistent APIs. What follows is the set of design principles governing the APIs which follow:
+
+1.  **Batching**: if an operation is explicitly defined in terms of matrices (i.e., two-dimensional arrays), then the associated interface should support "batching" (i.e., the ability to perform the operation over a "stack" of matrices). Example operations include:
+
+    -   `inv`: computing the multiplicative inverse of a square matrix.
+    -   `cholesky`: performing Cholesky decomposition.
+    -   `matmul`: performing matrix multiplication.
```

Comment:
The unfortunate absence of a matrix transpose again comes to mind. May be noted explicitly here perhaps, so that's a gap that may be filled in libraries and then added in the future.

Reply:
I could be missing something, but batch matrix transpose can be achieved now via the current `transpose` API by providing an axes permutation in which only the last two dimensions are swapped. As in TF, a `matrix_transpose` method supporting batching could be defined as such and simply be shorthand.